cmd/cli/app/auth/offline_use.go

```diff
@@ -0,0 +1,107 @@
+	"github.com/stacklok/minder/internal/util"
+	"github.com/stacklok/minder/internal/util/cli"
+)
+
+// offlineTokenUseCmd represents the offline-token use command
+var offlineTokenUseCmd = &cobra.Command{
+	Use:   "use",
+	Short: "Use an offline token",
+	Long: `The minder auth offline-token use command project lets you install and use an offline token
+for the minder control plane.
+
+Offline tokens are used to authenticate to the minder control plane without
+requiring the user's presence. This is useful for long-running processes
+that need to authenticate to the control plane.`,
+
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		clientConfig, err := config.ReadConfigFromViper[clientconfig.Config](viper.GetViper())
+		if err != nil {
+			return cli.MessageAndError("Unable to read config", err)
+		}
+
+		f := cmd.Flags().Lookup("file").Value.String()
+		fromEnv := cmd.Flags().Lookup("from-env").Value.String() == "true"
+		env := cmd.Flags().Lookup("env-var").Value.String()
+		if f == "" && !fromEnv {
+			return cli.MessageAndError("file or from-env is required", errors.New("file or from-env is required"))
+		} else if env == "" && fromEnv {
+			return cli.MessageAndError("env-var is required", errors.New("env-var is required"))
+		}
+
+		var tok string
+		if fromEnv {
+			var ok bool
+			tok, ok = os.LookupEnv(env)
+			if !ok {
+				return cli.MessageAndError("env var is required", fmt.Errorf("env var %s needs to be set", env))
+			}
+		} else {
+			tokbytes, err := os.ReadFile(f)
+			if err != nil {
+				return cli.MessageAndError("error reading file", fmt.Errorf("error reading file %s: %v", f, err))
+			}
+
+			tok = string(tokbytes)
+		}
+
+		// No longer print usage on returned error, since we've parsed our inputs
+		// See https://github.com/spf13/cobra/issues/340#issuecomment-374617413
+		cmd.SilenceUsage = true
+
+		issuerUrlStr := clientConfig.Identity.CLI.IssuerUrl
+		clientID := clientConfig.Identity.CLI.ClientId
+
+		creds, err := util.RefreshCredentials(tok, issuerUrlStr, clientID)
+		if err != nil {
+			return cli.MessageAndError("unable to refresh credentials", err)
+		}
+
+		// save credentials
+		filePath, err := util.SaveCredentials(util.OpenIdCredentials{
+			AccessToken:          creds.AccessToken,
+			RefreshToken:         creds.RefreshToken,
+			AccessTokenExpiresAt: creds.AccessTokenExpiresAt,
+		})
+		if err != nil {
+			cmd.PrintErrf("couldn't save credentials: %s\n", err)
+		}
+
+		cmd.Printf("Your access credentials have been saved to %s\n", filePath)
+
+		return nil
+	},
+}
+
+func init() {
+	OfflineTokenCmd.AddCommand(offlineTokenUseCmd)
+
+	offlineTokenUseCmd.Flags().StringP("file", "f", "offline.token", "The file that contains the offline token")
+	offlineTokenUseCmd.Flags().BoolP("from-env", "e", false, "Use the offline token from the environment")
+	offlineTokenUseCmd.Flags().StringP("env-var", "v", "MINDER_OFFLINE_TOKEN", "The environment variable to use for the offline token")
```

Comment:
Perhaps there's no need to have an explicit env var and another flag that says to read from it. Instead we can simplify this to have 2 flags:
* One that says read the token from this path(`--tokenpath offline.token`
* Another that says here's the token, i.e. `--token $MY_TOKEN_VAR`
* And the two should exclude each other.